querqy-for-lucene/querqy-lucene/src/main/java/querqy/lucene/contrib/rewrite/wordbreak/ClassicWordBreakCompoundRewriterFactory.java

```diff
@@ -62,7 +59,8 @@ public ClassicWordBreakCompoundRewriterFactory(final String rewriterId,
                                                    final List<String> reverseCompoundTriggerWords,
                                                    final boolean alwaysAddReverseCompounds,
                                                    final int maxDecompoundExpansions,
-                                                   final boolean verifyDecompoundCollation) {
+                                                   final boolean verifyDecompoundCollation,
+                                                   final List<String> protectedWords) {
```

Comment:
Strictly spoken, we would probably want to have sets instead of lists for both reverseCompoundTriggerWords (which has been around for a bit) and protectedWords. I think it doesn't really matter as we will deprecate and remove this rewriter soon in favour of the WordBreakCompoundRewriter(morphology=DEFAULT)

Reply:
Semantically it could be `Collection<String>` or even `Iterable<String>` as anything this factory does is iterating over the Strings to build a new object. I would only create `Set<String>` if we strictly *need* Set semantics. Otherwise, we can omit the overhead of creating an intermediary object and the slightly slower Set iteration performance.